pk/frontend.h

```diff
@@ -32,4 +32,36 @@ struct frontend_stat {
   uint32_t __unused5;
 };
 
+struct frontend_statx_timestamp {
+    int64_t  tv_sec;
+    uint32_t tv_nsec;
+    int32_t  __reserved;
+};
+
+struct frontend_statx {
+    uint32_t mask;
+    uint32_t blksize;
+    uint64_t attributes;
+    uint32_t nlink;
+    uint32_t uid;
+    uint32_t gid;
+    uint16_t mode;
+    uint16_t __spare0[1];
+    uint64_t ino;
+    uint64_t size;
+    uint64_t blocks;
+    uint64_t attributes_mask;
+    struct frontend_statx_timestamp atime;
+    struct frontend_statx_timestamp btime;
+    struct frontend_statx_timestamp ctime;
+    struct frontend_statx_timestamp mtime;
+    uint32_t rdev_major;
+    uint32_t rdev_minor;
+    uint32_t dev_major;
+    uint32_t dev_minor;
+    uint64_t mnt_id;
+    uint64_t __spare2;
+    uint64_t __spare3[12];
+};
+
```

Comment:
I think you can delete all of this and just have a macro `FRONTEND_STATX_SIZE`.  The contents of the structure don't really matter; pk only needs to know how big it is.

Reply:
Fixed. :)